### PR TITLE
complete task 8

### PR DESCRIPTION
**Pull request by AmadReihun**
complete task 8

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -311,5 +311,120 @@ describe("app", () => {
...
+    });
+
+    test("status code: 200 and returns article with the updated votes when the vote input is negative", () => {
+      return request(app).patch('/api/articles/1')
+      .send({ inc_votes: -50 })
+      .expect(200)
+        .then((response) => {
+          expect(response.body.updatedArticle.votes).toBe(50);
+        });
+    });
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:351]: Nice to check both increase and decrease of votes!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -311,5 +311,120 @@ describe("app", () => {
...
+
+    test("to get error 404 if the path is wrong", () => {
+      return request(app)
+        .patch('/api/wrongpath/1')
+        .expect(404)
+    });
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:357]: If you have the above test already for other endpoints you don't need it, it is not specific to the articles endpoint it is catching anything that does not match. You can have this test once and in it's own describe block to everything else 😄

File: __tests__/app.test.js [hunk 3]
```diff
@@ -311,5 +311,120 @@ describe("app", () => {
...
+    test("to get error 404 and respond with appropriate message when given wrong key name", () => {
+      return request(app)
+        .patch("/api/articles/150")
+        .send({votessss : 20})
+        .expect(404)
+        .then(({body}) => {
+          const message = body.msg;
+          expect(message).toBe("Not Found")
+        })
+    });
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:423]: superb error testing!

File: __tests__/app.test.js (hunk 2 above)
Comment (AmadReihun) [reply __tests__/app.test.js:357]: :D sure

File: __tests__/app.test.js (hunk 3 above)
Comment (AmadReihun) [reply __tests__/app.test.js:423]: Thanks :)

**Pull request merged**